### DISCUSSION TITLE
2023.7.8.0

### DISCUSSION THREAD

**Discussion by BornToBeRoot** [BornToBeRoot/NETworkManager#2404] [Releases]
[:page_facing_up: Changelog](https://github.com/BornToBeRoot/NETworkManager/blob/ba07cdbd6c9300d3406b1b3384771fc01a26a2e3/docs/Changelog/next-release.md)
[:book: Documentation](https://borntoberoot.net/NETworkManager/Documentation/Application)

---

| File | Checksum [SHA256]
|---|---|
|[Setup](https://github.com/BornToBeRoot/NETworkManager/releases/download/2023.7.8.0/NETworkManager_2023.7.8.0_Setup.exe)| `05FC29A3895826A59E4B0AF4EB5D27466308823FDDBCF19A51E5F31E6414413C` |
|[Portable](https://github.com/BornToBeRoot/NETworkManager/releases/download/2023.7.8.0/NETworkManager_2023.7.8.0_Portable.zip)| `440D8F08D784F9A2B2B58CAB24076F0AA735DF550DD9970ECB4061D3CB5A85FC` |
|[Archive](https://github.com/BornToBeRoot/NETworkManager/releases/download/2023.7.8.0/NETworkManager_2023.7.8.0_Archive.zip)| `EF6BCDEFEDA23150F8CA2F0292E512E4D5F00BDADA3CDD1BD308B04C9D92C721` |

**System requirements**
- Windows 10 / Server x64 (1809 or later)
- [.NET Desktop Runtime 6.x (LTS)](https://dotnet.microsoft.com/download/dotnet/6.0)

<hr /><em>This discussion was created from the release <a href='https://github.com/BornToBeRoot/NETworkManager/releases/tag/2023.7.8.0'>2023.7.8.0</a>.</em>

**Comment by Hammerfest** [comment]
Didn't want to file an issue until I was sure I wasn't missing something but documentation is mum on details to tell me I am wrong or its a bug

IP Scanner, the Hostname field, is there a reason it doesnt resolve for local systems? In your example in documentation on the website it also doesnt show local hostnames.

Any other IP Scanner I use resolves/displays hostnames but NETworkManager does not.

(Portable Version if thats related but think not due to documentation screenshot showing the same)

**Reply by BornToBeRoot** [reply]
Hi, for me it's working:

![image](https://github.com/BornToBeRoot/NETworkManager/assets/16019165/32df267c-0556-4d84-a4b4-0de1ee8b5e79)

You can check if the PTR record is set correctly on the DNS server:

```
> nslookup
> server <DNS_SERVER_IP>
> <IP_ADDRESS>
```

![image](https://github.com/BornToBeRoot/NETworkManager/assets/16019165/3fe1f0ea-1a0c-4a36-bca9-6d545f001338)


You can also try to enable this to show the error message:

![image](https://github.com/BornToBeRoot/NETworkManager/assets/16019165/dab9a3f8-d4d7-4078-b12e-2e6120a2d252)

**Comment by Hammerfest** [comment]
Yep, thats set, absolutely nothing.
Your results also don't look like just hostnames either (for windows with its 15 character limit) but a FQDN on something other then a basic network.

Images to follow

![Hostname-001](https://github.com/BornToBeRoot/NETworkManager/assets/396831/00ea49f5-cf89-4f1f-9e57-20f807257423)
![Hostname-002](https://github.com/BornToBeRoot/NETworkManager/assets/396831/1ccd34c2-31b2-44bf-bb1c-642a7912d573)

Images added, as you can see NETworkManager sees none of the hostnames (names field in advancedipscanner) but others do (AngryIP scanner and one whos name i dont recall for 3 off the top)

**Reply by BornToBeRoot** [reply]
The 16 chars limitation is only for Windows and the NETBIOS implementation... Maybe the Advanced IP Scanner checks the NETBIOS protocol instead of DNS (or as fallback)...

DNS can have 63 chars for the hostname and max 253 chars for the FQDN (See https://datatracker.ietf.org/doc/html/rfc1035#section-2.3.4).

Does your client resolve the hostname with nslookup? 

Example:
```
nslookup 192.168.x.x
```

**Reply by Hammerfest** [reply]
this does seem to be the case, nslookup returns no results registered

This is both on my home network and lab network at work so something is in play to keep registration off.

Ill make a feature request for netbios fallback
